### ISSUE TITLE
Wrong origin calculation in incident field for focus position outside of domain

### ISSUE THREAD

**Issue by pordyna** [ComputationalRadiationPhysics/picongpu#4209]
The incident field allows one to set the laser focus outside the Huygens surfaces, but the origin (of the laser coordinate system) calculation is assuming the focus is inside the box made by the generating surfaces. With that,  in some cases the origin will be calculated not to be somewhere on the surface of this box but rather on one of the generating surfaces extended into infinity (possibly somewhere far away from the simulation). This was already discussed offline in more detail with @sbastrakov

**Comment by pordyna** [comment]
oh @sbastrakov, actually I'm not sure if this is only for focus outside the domain.
So I was playing around with your `getOrigin` implementation in a cling jupyter notebook for my simulation parameters and
I get the origin outside the domain for a focus inside it.

So for: 
```c++
constexpr int32_t POSITION[3][2] = { {0, 0}, {16, -16}, {0, 0} };
constexpr double delta_x = 9.428090415820634e-9 / 4.0;
const double cellSize[3] = {delta_x, delta_x, delta_x };

const int globalDomainCells[3] = {960, 1728, 960};

// 45 degrees in x negative direction and y positive directon
const double direction[3] = {-0.7071067811865475, 0.7071067811865475, 0};
// focus
const double focus[3] = {1.0e-6, 1.0e-6, 1.6e-6};
auto firstIntersectionP = std::numeric_limits<double>::infinity();
for(uint32_t axis = 0u; axis < 3u; ++axis)
{
    // The expressions should generally work anyways, but to avoid potential 0/0
    if(std::abs(direction[axis]) > std::numeric_limits<double>::epsilon())
    {
        // Take into account 0.75 cells inwards shift of Huygens surface
        auto const minPosition
            = (static_cast<double>(POSITION[axis][0]) + 0.75) * cellSize[axis];
        firstIntersectionP
            = std::min(firstIntersectionP, (minPosition - focus[axis]) / direction[axis]);
        auto const maxPositionIdx = (POSITION[axis][1] > 0)
            ? POSITION[axis][1]
            : globalDomainCells[axis] + POSITION[axis][1];
        auto const maxPosition
            = (static_cast<double>(maxPositionIdx) - 0.75) * cellSize[axis];
        firstIntersectionP
            = std::min(firstIntersectionP, (maxPosition - focus[axis]) / direction[axis]);
    }
}
const double origin[3] = {focus[0] + firstIntersectionP * direction[0],
                          focus[1] + firstIntersectionP * direction[1],
                          focus[2] + firstIntersectionP * direction[2]};
```
I get origin equal  `{ 2.2609739e-06, -2.6097393e-07, 1.6000000e-06 }`

`const double focus[3] = {1.0e-6, 1.0e-6, 1.6e-6};` is inside the simulation box
```c++
double box_size[3] = {960 * delta_x, 1728 * delta_x, 960 * delta_x};
box_size
{ 2.2627417e-06, 4.0729351e-06, 2.2627417e-06 }
```
also inside the "laser" box considering `constexpr int32_t POSITION[3][2] = { {0, 0}, {16, -16}, {0, 0} };`

and still  `{ 2.2609739e-06, -2.6097393e-07, 1.6000000e-06 }` is outside the box.

**Comment by sbastrakov** [comment]
Thanks for investigating, I will look into these cases today or tomorrow.

**Comment by sbastrakov** [comment]
Solved with #4221 .
